polyply/src/simple_seq_parsers.py

```diff
@@ -0,0 +1,236 @@
+# Copyright 2022 University of Groningen
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from collections import OrderedDict
+from functools import partial
+import json
+import networkx as nx
+from networkx.readwrite import json_graph
+from vermouth.parser_utils import split_comments
+from vermouth.log_helpers import StyleAdapter, get_logger
+
+LOGGER = StyleAdapter(get_logger(__name__))
+
+ONE_LETTER_DNA = {"A": "DA",
+                  "C": "DC",
+                  "G": "DG",
+                  "T": "DT"}
+
+ONE_LETTER_RNA = {"A": "A",
+                  "C": "C",
+                  "G": "G",
+                  "T": "U"}
+
+def _monomers_to_linear_nx_graph(monomers):
+    """
+    Take a list of monomers and generate a linear
+    graph from them setting both the resid and
+    resname attribute.
+    """
+    seq_graph = nx.Graph()
+    mon_range = range(0, len(monomers))
+    seq_graph.add_edges_from(zip(mon_range[:-1], mon_range[1:]))
+    nx.set_node_attributes(seq_graph, {node: resname for node, resname in zip(seq_graph.nodes, monomers)}, "resname")
+    nx.set_node_attributes(seq_graph, {node: node+1 for node in seq_graph.nodes}, "resid")
+    return seq_graph
+
+def parse_plain_delimited(filehandle, delimiter=" "):
+    """
+    Parse a plain delimited text file. The delimiter can
+    be any special character.
+    """
+    with open(filehandle) as file_:
+        lines = file_.readlines()
+
+    monomers = []
+    for line in lines:
+        for resname in line.strip().split(delimiter):
+            monomers.append(resname)
+    seq_graph =  _monomers_to_linear_nx_graph(monomers)
+    return seq_graph
+
+parse_csv = partial(parse_plain_delimited, delimiter=',')
```

Comment:
can you strip the quotation marks to allow for both and/or mixed format?

Reply:
```python
if len(resname) >=2 and resname[0] == resname[-1] and resname[0] in ("'", '"'):
    resname = resname[1:-1]
```

This would work assuming the delimiter and spaces are invalid characters in resnames. Otherwise, you will need to account for escape characters as well and you are better off with importing a third party csv parser.

Reply:
and here I thought csv was simple; I'll see if there is a good lightweight csv parser

Reply:
So I had a loot at the python standard library for csv and they simple implement different language types like unix, excel etc. I'm not sure I want to do this, because that seems pretty complicated as you'd have to set the language type through CLI and cannot identify it automatically. I'm thinking to remove csv support. @jbarnoud do you have a better idea?